k2/python/csrc/torch/v2/any.cu

```diff
@@ -70,15 +70,47 @@ void PybindRaggedAny(py::module &m) {
 
   any.def(
       "__getitem__",
-      [](RaggedAny &self, int32_t i) -> RaggedAny {
-        return self.Index(/*axis*/ 0, i);
+      [](RaggedAny &self, int32_t i) -> py::object {
+        if (self.any.NumAxes() > 2) {
+          RaggedAny ragged = self.Index(/*axis*/ 0, i);
+          return py::cast(ragged);
+        } else {
+          K2_CHECK_EQ(self.any.NumAxes(), 2);
+          const int32_t *row_split1_data = self.any.RowSplits(1).Data();
+          int32_t begin = row_split1_data[i],
+                  end = row_split1_data[i + 1];
+          Dtype t = self.any.GetDtype();
+          FOR_REAL_AND_INT32_TYPES(t, T, {
+            Array1<T> array =
+                self.any.Specialize<T>().values.Arange(begin, end);
+            torch::Tensor tensor = ToTorch(array);
+            return py::cast(tensor);
+          });
+        }
+        // Unreachable code
+        return py::none();
       },
       py::arg("i"), kRaggedAnyGetItemDoc);
 
+  any.def(
+      "__getitem__",
+      [](RaggedAny &self, const py::slice &slice) -> RaggedAny {
+        py::ssize_t start = 0, stop = 0, step = 0, slicelength = 0;
+        if (!slice.compute(self.any.Dim0(), &start, &stop, &step, &slicelength))
+          throw py::error_already_set();
+        int32_t istart = static_cast<int32_t>(start);
+        int32_t istop  = static_cast<int32_t>(stop);
+        int32_t istep  = static_cast<int32_t>(step);
+        K2_CHECK_EQ(istep, 1) << "Only support slicing with step 1, given : "
+                              << istep;
+
+        return self.Arange(/*axis*/ 0, istart, istop);
+      }, kRaggedAnyGetItemSliceDoc);
+
   any.def("index",
           static_cast<RaggedAny (RaggedAny::*)(RaggedAny &, bool)>(
```

Comment:
The signature also needs to be changed. Otherwise, it won't compile, I think.

Reply:
Wondering how did you get the jupyter notebook working as the code won't compile.